addons/advanced_fatigue/functions/fnc_calculateBodyTemperature.sqf

```diff
@@ -0,0 +1,73 @@
+/*
+* Author: Pasrules
+* Calculates body temperature in degC, normal = 37
+*
+* Arguments:
+* 0: bodyTemperature <NUMBER>
+*
+* Return Value:
+* bodyTemperature <NUMBER>
+*
+*
+* Example
+* [37] call ace_advanced_fatigue_fnc_calculateBodyTemperature
+*/
+
+#include "script_component.hpp"
+params ["_bodyTemp"];
+
+if (["ACE_Weather"] call EFUNC(common,isModLoaded)) then {
+
+    //Weather
+    private _playerAltitude = (getPosASL ACE_player) select 2;
+    private _temperature = _playerAltitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _humidity = humidity;
+    private _barometricPressure = _playerAltitude call EFUNC(weather,calculateBarometricPressure);
+    private _airDensity = [_temperature, _barometricPressure, _humidity] call EFUNC(weather,calculateAirDensity);
+    private _windSpeed = [eyePos ACE_player, false, false, false] call EFUNC(weather,calculateWindSpeed);
+    private _windFactor = linearConversion [0, 20, _windSpeed, 0, 1, true];
+
+    //Average male [kg, cm, degC]
+    private _male = [87,177,37];
+    private _bodyWeight = _male select 0;
+    private _bodyHeight = _male select 1;
+    private _normalTemp = _male select 2;
+    private _skin = 0.5; //skin conductance factor
+    private _capacity = _skin / (_bodyWeight * 4.18); //specific heat capacity of the body
+    private _surfaceArea = sqrt (_bodyWeight * _bodyHeight / 3600); //BodySA = SQRT(kg*cm/3600)
+	
+
+    //Environment
+    private _sun = 120 * (1 - overcast);
+    private _air = (_temperature - _normalTemp) * 2.718 ^ _windFactor * _airDensity;
```

Comment:
The weather module provides helper functions to calculate the [wind chill](https://github.com/acemod/ACE3/blob/master/addons/weather/functions/fnc_calculateWindChill.sqf) and [heat index](https://github.com/acemod/ACE3/blob/master/addons/weather/functions/fnc_calculateHeatIndex.sqf). Maybe you have use for them.